src/ext/components/partial-dependence-input.jade

```diff
@@ -11,20 +15,55 @@
       tr
         th
           label Model:
-        td
+        td(colspan='2')
           select(data-bind="options:models, value:selectedModel, optionsCaption:'(Select)'")
       tr
         th
           label Frame:
-        td
-          select(data-bind="options:frames, value:selectedFrame, optionsCaption:'(Select)'")
+        td(colspan='2')
+          select(data-bind="options:frames, value:selectedFrame, optionsCaption:'(Select)', event:{ change:updateColumns}")
       tr
         th nbins
         td
           input.flow-textbox(type='text' data-bind="value:nbins")
         td How many levels should PDP compute. More levels will make it slower.
       tr
         th
+          label Select columns?
+        td 
+          input(type='checkbox' data-bind='checked:useCustomColumns')
+        td Checking this will allow you to select custom columns for PDP. By default, the top 10 features are used. Those features are sorted by variable importance.
+      tr(data-bind='visible:useCustomColumns')
+        th 
+          label Available columns:
+        td(colspan='2')
+          div(style='height:25px')
+            input.flow-textbox(type='text' placeholder='Search...' data-bind="value:searchTerm, valueUpdate:'afterkeydown'")
+          div(style='font-size:80%' data-bind='text:searchCaption')
+          div(style='margin-bottom:5px;height:300px;overflow:auto;border:1px dotted #ccc')
+            table
+              tbody
+                // ko foreach:visibleItems
+                tr
+                  td
+                    label
+                      input(type='checkbox' data-bind='checked:isSelected')
+                      span(data-bind='text:value')
+                  td.flow-narrow.flow-aside(data-bind='text:type')
+                  td.flow-narrow.flow-aside(data-bind='text:missingLabel')
+                // /ko
+
+          table(style='margin-bottom:10px;width:100%')
+            tbody
+              tr
+                td
+                  +button('All', 'check-square-o', 'selectFiltered', 'hasFilteredItems')
```

Comment:
button syntax: 
first value is text that appears on the button.  
second value is the fa font-awesome css class of the svg glyph to display as the button
third value is the function that will be called when the button is clicked
fourth value is the the function that will return the button's state - is the button grayed out or not (enabled or not) (could be red or not or hidden or not. in our case it is grayed out).